src/shaders/cs_gaussian_blur_horizontal.comp

```diff
@@ -27,11 +27,26 @@ void main()
     uvec2 coord = uvec2(gl_GlobalInvocationID.x, gl_GlobalInvocationID.y);
     vec2 pos = coord;
 
-    vec3 outputValue;
-    if ((c_ycbcrMask & 1) != 0)
-        outputValue = gaussian_blur(s_ycbcr_samplers[0], pos, c_blur_radius, false, false).rgb;
-    else
-        outputValue = gaussian_blur(s_samplers[0], pos, c_blur_radius, false, true).rgb;
+    vec3 outputValue = vec3(0);
+
+    if (c_layerCount > 0) {
+        if ((c_ycbcrMask & 1) != 0)
+            outputValue = gaussian_blur(s_ycbcr_samplers[0], 0, pos, c_blur_radius, false, false).rgb;
+        else
+            outputValue = gaussian_blur(s_samplers[0], 0, pos, c_blur_radius, false, true).rgb;
+    }
+
+    for (int i = 1; i < c_layerCount; i++) {
+        vec4 layerColor;
+        if ((c_ycbcrMask & 1) != 0)
+            layerColor = gaussian_blur(s_ycbcr_samplers[i], i, pos, c_blur_radius, false, false);
+        else
+            layerColor = gaussian_blur(s_samplers[i], i, pos, c_blur_radius, false, true);
+
+        float opacity = u_opacity[i];
+        float layerAlpha = opacity * layerColor.a;
+        outputValue = layerColor.rgb * opacity + outputValue * (1.0f - layerAlpha);
```

Comment:
The horizontal pass works in srgb space, so this blending is wrong.

Reply:
probably best to switch it to linear for bgra images and to do srgbToLinear after the blur for nv12

Reply:
Reworked the srgb stuff a bit.